### ISSUE TITLE
Modifying crop orientation prediction model classes does not result in appropriate rectification

### ISSUE THREAD

**Issue by milosacimovic** [mindee/doctr#1815]
### Bug description

Expecting that changing classes for the crop orientation model would be reflected in how the rectification process is done. 
After inspection it seems that the rectification is being done not based on classes but based on indices in the classes array.
Does this make sense to you? What's the justification for this? 

My specific case is that I know I won't have any 180 oriented text so why sacrifice the accuracy for something I know won't occurr.

### Code snippet to reproduce the bug

```python
orient_arch = orient_arch(orient_onnx_path)
orient_arch.cfg["classes"] = [0, -90, 0, 90]
```

### Error traceback

There's no error per-se but a crop which was mispredicted as having an orientation of 180 still gets rotated incorrectly by 180 while I have set explicitly the class to be 0 as I don't expect any 180 oriented text.

### Environment

DocTR version: v0.8.1
TensorFlow version: 2.14.0
PyTorch version: 2.1.1+cpu (torchvision 0.16.1+cpu)
OpenCV version: 4.10.0
OS: Microsoft Windows 10 Business
Python version: 3.11.9
Is CUDA available (TensorFlow): No
Is CUDA available (PyTorch): No
CUDA runtime version: 12.1.105
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect

### Deep Learning backend

is_tf_available: True
is_torch_available: True

**Comment by felixdittrich92** [comment]
Hi @milosacimovic :wave:,

That's correct the classes are only to display later in output nothing else :)

We need for `np.rot90` and `np.roll` the indices so how many times we have to rotate

refs.:

https://github.com/mindee/doctr/blob/48873e005ecf5f57c55789d83d71ade9aef1457e/doctr/models/_utils.py#L117

https://github.com/mindee/doctr/blob/48873e005ecf5f57c55789d83d71ade9aef1457e/doctr/models/classification/predictor/pytorch.py#L61

**Comment by milosacimovic** [comment]
Yes, that's fine. Thank you @felixdittrich92  once again
